### DISCUSSION TITLE
Seeking guidance on external clock for sensor synchronization in TB4

### DISCUSSION THREAD

**Discussion by DavidY91** [turtlebot/turtlebot4#345] [Q&A]
Hi all, I would like to use external clock to trigger multiple sensors(e.g., camera/lidar/imu/wheels) so that the sensors commence measurements simultaneously. I'm curious if TB4 supports such synchronization approach? 
Any helpful discussions are welcome and appreciated！

**Comment by bmaxdk** [comment]
Have you determine if TB4 sensors have input pins for external triggers or if they can be synchronized via software? If the sensors support hardware triggering, identify the appropriate GPIO pins on the TurtleBot4’s microcontroller or single-board computer (e.g., Raspberry Pi or NVIDIA Jetson) that can be used to input the external clock signal. You may want to try  GPIO pins.
